Example/ViewController.swift

```diff
@@ -54,20 +54,44 @@ class ViewController: UIViewController {
             }
         }
     }
+    
+    var currentRouteIndex = 0 {
+        didSet {
+            showCurrentRoute()
+        }
+    }
 
+    func showCurrentRoute() {
+        guard let currentRoute = currentRoute else { return }
+        
+        var routes = [currentRoute]
+        routes.append(contentsOf: self.routes!.filter {
+            $0 != currentRoute
+        })
```

Comment:
This would be a bit easier to understand:

```swift
guard var prioritizedRoutes = routes else { return }
prioritizedRoutes.insert(prioritizedRoutes.remove(at: currentRouteIndex), at: 0)
```